plugins/promela/hu.bme.mit.gamma.xsts.promela.transformation/src/hu/bme/mit/gamma/xsts/promela/transformation/util/Namings.xtend

```diff
@@ -32,9 +31,37 @@ class Namings {
 	
 	static def String customizeEnumLiteralName(EnumerationLiteralExpression expression) '''«expression.reference.typeDeclaration.name»«expression.reference.name»'''
 	static def String customizeEnumLiteralName(EnumerationTypeDefinition type, EnumerationLiteralDefinition literal) '''«type.typeDeclaration.name»«literal.name»'''
-	static def String customizeEnumLiteralName(State state, Region parentRegion, ComponentInstanceReferenceExpression instance) '''«parentRegion.name.regionTypeName»_«parentRegion.containingComponent.FQNUpToComponent»«state.customizeName»'''
-	static def String customizeEnumLiteralName(State state, Region parentRegion, SynchronousComponentInstance instance) '''«parentRegion.name.regionTypeName»_«parentRegion.containingComponent.FQNUpToComponent»«state.customizeName»'''
+	static def String customizeEnumLiteralName(State state) '''«state.parentRegion.name.regionTypeName»_«state.parentRegion.containingComponent.FQNUpToComponent»«state.customizeName»''' 
+	static def String customizeEnumLiteralName(Region region, String literal) '''«region.name.regionTypeName»_«region.containingComponent.FQNUpToComponent»«literal»'''
+	static def String customizeEnumLiteralName(State state, Region parentRegion) '''«parentRegion.name.regionTypeName»_«parentRegion.containingComponent.FQNUpToComponent»«state.customizeName»'''
+	
+	static def createEnumMapping(List<Region> regions) {
```

Comment:
Do we actually need this? Are the state literals not back-annotated correctly based on `getSingleTargetStateName`?